packages/MJCore/src/generic/baseEntity.ts

```diff
@@ -342,6 +342,12 @@ export abstract class BaseEntity {
         return this.EntityInfo.PrimaryKeys.map(pk => this.GetFieldByName(pk.Name));
     }
 
+    get CompositeKey(): CompositeKey {
```

Comment:
@JS-BC  wrong way to do this. Why construct a composite key each time it is requested. ALSO, remember what I told you, over time, CK might have MORE information than just the array of PKeys in it. You need to not take shortcuts like this. The native representation within this class must be clean and correct and be a composite key and we should be using that everywhere. You could make the argument that we should still expose PrimaryKeys as an array for back compat, but I'd rather not even do that. If you were to do that, you'd do it exactly the opposite way, NATIVE way to store data is a CompositeKey inside BaseEntity, and you would create a GETTER that would pull out the internal PKVals array and pass it along, but it is not clean.